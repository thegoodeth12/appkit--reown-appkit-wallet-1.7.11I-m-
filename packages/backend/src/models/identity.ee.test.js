import { describe, it, expect } from 'vitest';
import Identity from '@/models/identity.ee.js';
import User from '@/models/user.js';
import SamlAuthProvider from '@/models/saml-auth-provider.ee.js';
import Base from '@/models/base.js';

describe('Identity model', () => {
  it('tableName should return correct name', () => {
    expect(Identity.tableName).toBe('identities');
  });

  it('jsonSchema should have correct validations', () => {
    expect(Identity.jsonSchema).toMatchSnapshot();
  });

  it('relationMappings should return correct associations', () => {
    const relationMappings = Identity.relationMappings();

    const expectedRelations = {
      user: {
        relation: Base.BelongsToOneRelation,
        modelClass: User,
        join: {
          from: 'users.id',
          to: 'identities.user_id',
        },
      },
      samlAuthProvider: {
        relation: Base.BelongsToOneRelation,
        modelClass: SamlAuthProvider,
        join: {
          from: 'saml_auth_providers.id',
          to: 'identities.provider_id',
        },
      },
    };

    expect(relationMappings).toStrictEqual(expectedRelations);
  });
});
