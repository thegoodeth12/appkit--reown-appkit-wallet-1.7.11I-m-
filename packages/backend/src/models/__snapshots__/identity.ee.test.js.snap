// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Identity model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "providerId": {
      "format": "uuid",
      "type": "string",
    },
    "providerType": {
      "enum": [
        "saml",
      ],
      "type": "string",
    },
    "remoteId": {
      "minLength": 1,
      "type": "string",
    },
    "userId": {
      "format": "uuid",
      "type": "string",
    },
  },
  "required": [
    "providerId",
    "remoteId",
    "userId",
    "providerType",
  ],
  "type": "object",
}
`;
